detox/android/detox/src/main/java/com/wix/detox/DetoxManager.java

```diff
@@ -213,14 +212,34 @@ public void run() {
 
     @Override
     public void onConnect() {
-        wsClient.sendAction("ready", Collections.emptyMap(), -1000L);
+        readyAction.perform("", -1000L);
     }
 
     @Override
     public void onClosed() {
         stop();
     }
 
+    private void initReactNativeIfNeeded() {
+        if (ReactNativeSupport.isReactNativeApp()) {
+            ReactNativeCompat.waitForReactNativeLoad(reactNativeHostHolder);
```

Comment:
We need to think of a way to support native apps with no RN as dependency. Currently native apps crash in classNotFound when running a detox test .
But this is probably a different issue

Reply:
It's different but sounds important nonetheless. Do we have an actual issue? I don't understand why this would happen, though, because we only check for the `ReactApplication` class' existence